src/mpi/coll/iallgather/iallgather_intra_gentran_brucks.c

```diff
@@ -18,14 +18,13 @@
 
 int MPIR_Iallgather_intra_gentran_brucks(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
                                          void *recvbuf, int recvcount, MPI_Datatype recvtype,
-                                         MPIR_Comm * comm_ptr, MPIR_Request ** request)
+                                         MPIR_Comm * comm_ptr, int k, MPIR_Request ** request)
```

Comment:
It bothers me that the argument order is different here from `MPII_Gentran_Iallgather_intra_brucks` below. Can we make them consistent?

Personally I would prefer:  MPI layer arguments (as specified by MPI interface), then semantics related arguments (`request`, `sched`, `errFlag`), then algorithm specific parameters?